### PR TITLE
multicast: add source and player API calls

### PR DESCRIPTION
**Pull request by cHuberCoffee**
This adds some API calls which are useful for the exchangeable multicast source/player.

### CODE REVIEW

File: modules/multicast/player.c [hunk 1]
```diff
@@ -560,3 +560,22 @@ void mcplayer_stop(void)
 {
 	player = mem_deref(player);
 }
+
+
+/**
+ * Initialize everything needed for the player beforhand
+ *
+ */
+void mcplayer_init(void)
```
Comment (alfredh) [inline modules/multicast/player.c:569]: in general an init function should return `int` (errno) because it can potentially fail.

**Comment by alfredh** [comment]
this patch adds some empty functions ...

what is the purpose of that ?

**Comment by cHuberCoffee** [comment]
The multicast source and player are exchangeable. It may be the case that somebody want to implement its own source and player. The API calls are useful if somebody want to initialize one or both with something which must be accessible right at the beginning of the lifetime of the module. The standard implementation does not need it but i want to open the support for such sources and players.

**Comment by cspiel1** [comment]
Replacing of player and source needs only to patch `multicast/module.mk`. The file multicast.c can stay untouched. We have such alternative player and source in one of our products.

**Pull request merged**